storage/innobase/fil/fil0crypt.cc

```diff
@@ -3051,9 +3052,16 @@ void fil_crypt_set_rotation_iops(uint val) {
 Adjust encrypt tables
 @param[in]	val		New setting for innodb-encrypt-tables */
 
-void fil_crypt_set_encrypt_tables(enum_default_table_encryption val) {
+bool fil_crypt_set_encrypt_tables(enum_default_table_encryption val,
+                                  bool is_server_starting) {
+  // It is always OK to set default_table_encryption on server
+  // startup
+  if (!is_server_starting && (srv_n_fil_crypt_threads_requested != 0)) {
```

Comment:
do we need is_server_starting? Is this function called on startup? Wouldn't the check in update() function of the variable be sufficient?

Reply:
Yes, it is called on startup via handler - fix_default_table_encryption. We set srv_default_table_encryption only when srv_uuid became available. In this function we check that we can change default_table_encryption (i.e. number of encryption threads is equal to 0).